### PR TITLE
[HDInsight] - Swagger-based Job SDK

### PR DESCRIPTION
**Pull request by idear1203**
### Swagger PR Links
- [Azure/azure-rest-api-specs#4489: [[HDInsight] - Swagger based job SDK](https://github.com/Azure/azure-rest-api-specs/pull/4489)
- [Azure/azure-rest-api-specs#4883: [[HDInsight] - Change package according to dataplane standard](https://github.com/Azure/azure-rest-api-specs/pull/4883)

### Note
- This change aims to update [the existing Jobs SDK](https://github.com/Azure/azure-sdk-for-net/tree/master/src/ResourceManagement/HDInsightJob) to make it swagger-based
- Move the Management SDK under folder `Management` and make job SDK under folder `DataPlane`
- Update `generated.cmd` of Management SDK to point to the correct path since the Management SDK code path has changed

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd
Comment (shahabhijeet) [inline src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd]: you no longer need this file.
Rather you should be using the .PS1 file that you see in other SDKs.
Model your .ps1 by looking at any other SDK directory.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
All the packages that you are planning to publish under dataPlane should be named appropriately.
For management plane SDK, we follow the below guideline
Microsoft.Azure.Management.`<ServiceName>`.csproj
For dataplane it should be
Microsoft.Azure.`<ServiceName>`.csproj, so in your case your Dataplane SDK should be called Microsoft.Azure.HDInsight.csproj

File: src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd
Comment (idear1203) [reply src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd]: Hi @shahabhijeet , I will rename the .csproj file. But what about the namespace? We want to keep "Microsoft.Azure.Management.HDInsight" because it is from original [Hyak based Job SDK](https://github.com/Azure/azure-sdk-for-net/blob/master/src/ResourceManagement/HDInsightJob/HDInsightJob/HDInsightJob.csproj#L9-L10). Does it make sense to you? Could you please give us some suggestions on this topic?

Comment (idear1203) [reply src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd]: I will create .ps1 in the next iteration.

Comment (shahabhijeet) [reply src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd]: @idear1203 it is important you generate the SDK using the ps1, by doing so you will be able to generate SDK with the proper meta data.
Also renaming of the files is important to be done now. Unless you have a reason why it should be done next time and not now?

Comment (idear1203) [reply src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd]: Oh. I mean I will do now in the new commit. Just want to check if you think  "Microsoft.Azure.HDInsight.Job" is a good namespace?

Comment (shahabhijeet) [reply src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd]: @idear1203 is Microsoft.Azure.HDInsight.Job a resource provider or is it Microsoft.Azure HDInsight?

Comment (idear1203) [reply src/SDKs/HDInsight/Management/Management.HDInsight/generated.cmd]: No, it is not a resource provider, it is a wrapper of open source [Apache Templeton API](https://cwiki.apache.org/confluence/display/Hive/WebHCat+UsingWebHCat). We are moving from Hydra to Swagger. The codebase for the hydra spec that generates the Jobs SDK is [here](https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fhydra-specs-pr%2Ftree%2Fmaster%2FHDInsight2%2FMicrosoft.Azure.Management.HDInsight.Job.Specification&data=02%7C01%7CDongwei.Wang%40microsoft.com%7C53ce95cbabe143dfa92308d6345c6e8a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636753968976913095&sdata=i9A%2FF%2FdTv6W35u2YCpITthDucA7Ap6wlzfbC3BsZ%2FBY%3D&reserved=0) and the codebase for the generated Jobs SDK is [here](https://apac01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-sdk-for-net%2Ftree%2Fmaster%2Fsrc%2FResourceManagement%2FHDInsightJob%2FHDInsightJob&data=02%7C01%7CDongwei.Wang%40microsoft.com%7C53ce95cbabe143dfa92308d6345c6e8a%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636753968976913095&sdata=ZqTPGEHURqL4THNqoeH70BtBoyxZga2sYZmVzx7mwIg%3D&reserved=0).

We would like to name the package as "Microsoft.Azure.HDInsight.Job" rather than "Microsoft.Azure HDInsight", since all this package is related to job management and in the old Hydra-based SDK, the namespace is "Microsoft.Azure.Management.HDInsight.Job". We thought it would be nice if we could keep "Microsoft.Azure.Management.HDInsight.Job". But we think it makes sense if we need to use "Microsoft.Azure.HDInsight.Job". What do you think?

**Review by shahabhijeet — APPROVED** [review]
Will merge once @weshaggard and @AlexGhiondea can approve.

**Comment by AlexGhiondea** [comment]
How much of the code is autogenerated and how much of it is hand written?

**Comment by idear1203** [comment]
code under [Generated](https://github.com/idear1203/azure-sdk-for-net/tree/hdi_support_job_submission/src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Generated) are generated, while code under [Customizations](https://github.com/idear1203/azure-sdk-for-net/tree/hdi_support_job_submission/src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Customizations) are hand written. Other code are just [test code](https://github.com/idear1203/azure-sdk-for-net/tree/hdi_support_job_submission/src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job.Tests) or [project related files](https://github.com/idear1203/azure-sdk-for-net/tree/hdi_support_job_submission/src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job).

File: src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj
Comment (dsgouda) [inline src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj]: We recommend starting `preview` versions from `0.9.0-preview`

File: src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj
Comment (dsgouda) [inline src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj]: nit: fix indentation

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj
Comment (idear1203) [reply src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj]: @dsgouda , please notice this is not a new package. We are trying to move from old Hyak-based jobs SDK to Swagger-based jobs SDK. You can find the old package [here](https://www.nuget.org/packages/Microsoft.Azure.Management.HDInsight.Job/), its version is `2.0.7`. That is the reason we want to make version of this package starts from `3.0.0-preview`. Does it make sense?

File: src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj
Comment (idear1203) [reply src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj]: @dsgouda Thanks, I have fixed this issue.

File: src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj
Comment (dsgouda) [reply src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj]: Appreciate the info

**Review by dsgouda — APPROVED** [review]

**Comment by dsgouda** [comment]
@shahabhijeet merge when ready

File: src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj
Comment (idear1203) [reply src/SDKs/HDInsight/DataPlane/Microsoft.Azure.HDInsight.Job/Microsoft.Azure.HDInsight.Job.csproj]: Force pushed a new commit to put commits together.

**Review by AlexGhiondea — APPROVED** [review]
I skimmed through the data plane code -- thanks for working on this!

**Pull request merged**
